clients/unshaded/src/main/java/tachyon/client/block/LocalBlockOutStream.java

```diff
@@ -0,0 +1,164 @@
+/*
+ * Licensed to the University of California, Berkeley under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.client.block;
+
+import java.io.IOException;
+import java.io.RandomAccessFile;
+import java.nio.ByteBuffer;
+import java.nio.MappedByteBuffer;
+import java.nio.channels.FileChannel;
+
+import com.google.common.base.Preconditions;
+import com.google.common.io.Closer;
+
+import tachyon.Constants;
+import tachyon.client.ClientContext;
+import tachyon.util.io.BufferUtils;
+import tachyon.util.io.FileUtils;
+import tachyon.util.network.NetworkAddressUtils;
+import tachyon.worker.WorkerClient;
+
+/**
+ * Provides a streaming API to write to a Tachyon block. This output stream will directly write
+ * the input to a file in local Tachyon storage.
```

Comment:
As per Bin's comments, I think this class should also be marked as not thread safe.